crates/core_simd/src/vector.rs

```diff
@@ -160,12 +209,50 @@ where
         mask: Mask<isize, LANES>,
         idxs: Simd<usize, LANES>,
     ) {
-        // We must construct our scatter mask before we derive a pointer!
-        let mask = (mask & idxs.lanes_lt(Simd::splat(slice.len()))).to_int();
+        let mask: Mask<isize, LANES> = mask & idxs.lanes_lt(Simd::splat(slice.len()));
+        // SAFETY: We have masked-off out-of-bounds lanes.
+        unsafe { self.scatter_select_unchecked(slice, mask, idxs) }
+    }
+
+    /// Writes the values in a SIMD vector to multiple potentially discontiguous indices in `slice`.
+    /// Lanes masked `false` are not written.
```

Comment:
```suggestion
    /// If a mask is `false`, the lane is not written.
```

Reply:
Hmm. So I think some of these comments are good improvements and I would probably just adopt them directly, but this is revealing a kinda mental-view difference here that I want to dig into a bit:

I don't want to talk about "mask" for individual lanes because I think that's confusing: the way I am looking at it, the overall "data structure" is a `Mask` and it functionally has "lanes" of its own, even if they are a bit opaque or are only 1-bit-wide. So a `Mask` has multiple `true` and `false` values in it, and thus talking about it being either value _per se_ is... odd, to me.

Notionally:
`self` has lanes containing numeric values.
`mask` has lanes containing go/no-go values.
`idxs` has lanes containing indices.
The controlling structure is `{ value, index } & bool`, which one could call "masked lanes" or "masked indices", depending on how you look at it.

Reply:
I am now thinking, should we perhaps instead call the mask variable `enabled` or similar?  That encodes the write-on-`true` nature in the name better than simple `mask`.

Reply:
`enable`, as Rust is an imperative language. 👑